arches/management/commands/packages.py

```diff
@@ -335,7 +336,11 @@ def handle(self, *args, **options):
             self.setup(package_name, es_install_location=options["dest_dir"])
 
         if options["operation"] == "install":
-            self.install(package_name)
+            warnings.warn(
```

Comment:
This is very puzzling, but I don't get any warning output if the DeprecationWarning type is passed into the warn method. Any other type works fine, just not the deprecation type:

<img width="1067" alt="image" src="https://github.com/user-attachments/assets/f732d4ff-0ef6-4204-ad1c-a0d2a8ca1412">

Maybe @johnatawnclementawn can confirm?

Reply:
Ah, one of the weirdsies about Python is that `DeprecationWarning` is filtered out by default, I guess so that you have a longer runway to cope with pending changes.

I think run with python -Wd ... (default) or -Wa (all) to see it, but you have a point that we should probably change it to UserWarning.

Reply:
I can repro what Cyrus is seeing without `-Wa`:
```
(ENV76) johnathanclementi@Johnathans-MacBook-Pro arches-lingo % python manage.py packages -o install    
System check identified some issues:

WARNINGS:
?: (arches.W001) Cache backend does not support rate-limiting
        HINT: Your cache: django.core.cache.backends.dummy.DummyCache
        Supported caches: ('django.core.cache.backends.memcached.PyLibMCCache', 'django.core.cache.backends.memcached.PyMemcacheCache', 'django.core.cache.backends.redis.RedisCache')
?: (guardian.W001) Guardian authentication backend is not hooked. You can add this in settings as eg: `AUTHENTICATION_BACKENDS = ('django.contrib.auth.backends.ModelBackend', 'guardian.backends.ObjectPermissionBackend')`.
?: (staticfiles.W004) The directory '/Users/johnathanclementi/Documents/Arches/Projects/rdm/arches-lingo/arches_lingo/media/build' in the STATICFILES_DIRS setting does not exist.
?: (staticfiles.W004) The directory '/Users/johnathanclementi/Documents/Arches/Projects/rdm/arches-lingo/node_modules' in the STATICFILES_DIRS setting does not exist.
?: (urls.W005) URL namespace 'admin' isn't unique. You may not be able to reverse all URLs in this namespace
?: (urls.W005) URL namespace 'oauth2' isn't unique. You may not be able to reverse all URLs in this namespace
operation: install
2024-08-01 18:54:40,667 arches.app.utils.task_management ERROR    A celery broker is running, but a celery worker is not available
```

With `-Wa`:
```
(ENV76) johnathanclementi@Johnathans-MacBook-Pro arches-lingo % python -Wa manage.py packages -o install
/Users/johnathanclementi/Documents/Arches/Projects/rdm/arches/arches/apps.py:21: DeprecationWarning: Providing boolean values to FILE_TYPE_CHECKING is deprecated. Starting with Arches 8.0, the only allowed options will be None, 'lenient', and 'strict'.
  warnings.warn(FILE_TYPE_CHECKING_MSG, DeprecationWarning)
/Users/johnathanclementi/Documents/Arches/Projects/rdm/ENV76/lib/python3.12/site-packages/edtf/parser/grammar.py:113: UserWarning: warn_multiple_tokens_in_named_alternation: setting results name 'lower' on Or expression will return a list of all parsed tokens in an And alternative, in prior versions only the first token was returned; enclose contained argument in Group
  level1Interval = l1Start("lower") + "/" + l1End("upper")
/Users/johnathanclementi/Documents/Arches/Projects/rdm/ENV76/lib/python3.12/site-packages/edtf/parser/grammar.py:210: UserWarning: warn_ungrouped_named_tokens_in_collection: setting results name 'ssn' on And expression collides with 'season' on contained expression
  ^ (season("ssn") + UASymbol("season_ua"))
/Users/johnathanclementi/Documents/Arches/Projects/rdm/ENV76/lib/python3.12/site-packages/edtf/parser/grammar.py:231: UserWarning: warn_multiple_tokens_in_named_alternation: setting results name 'lower' on Or expression will return a list of all parsed tokens in an And alternative, in prior versions only the first token was returned; enclose contained argument in Group
  level2Interval = (dateOrSeason("lower") + "/" + dateWithInternalUncertainty("upper")) \
/Users/johnathanclementi/Documents/Arches/Projects/rdm/ENV76/lib/python3.12/site-packages/edtf/parser/grammar.py:232: UserWarning: warn_multiple_tokens_in_named_alternation: setting results name 'upper' on Or expression will return a list of all parsed tokens in an And alternative, in prior versions only the first token was returned; enclose contained argument in Group
  ^ (dateWithInternalUncertainty("lower") + "/" + dateOrSeason("upper")) \
/Users/johnathanclementi/Documents/Arches/Projects/rdm/ENV76/lib/python3.12/site-packages/edtf/parser/grammar.py:242: UserWarning: warn_ungrouped_named_tokens_in_collection: setting results name 'lower' on And expression collides with 'month' on contained expression
  ^ (yearMonth("lower") + ".." + yearMonth("upper")) \
/Users/johnathanclementi/Documents/Arches/Projects/rdm/ENV76/lib/python3.12/site-packages/edtf/parser/grammar.py:242: UserWarning: warn_ungrouped_named_tokens_in_collection: setting results name 'upper' on And expression collides with 'month' on contained expression
  ^ (yearMonth("lower") + ".." + yearMonth("upper")) \
/Users/johnathanclementi/Documents/Arches/Projects/rdm/ENV76/lib/python3.12/site-packages/edtf/parser/grammar.py:278: UserWarning: warn_multiple_tokens_in_named_alternation: setting results name 'level0' on Or expression will return a list of all parsed tokens in an And alternative, in prior versions only the first token was returned; enclose contained argument in Group
  edtfParser = level0Expression("level0") ^ level1Expression("level1") ^ level2Expression("level2")
/Users/johnathanclementi/Documents/Arches/Projects/rdm/ENV76/lib/python3.12/site-packages/edtf/parser/grammar.py:278: UserWarning: warn_multiple_tokens_in_named_alternation: setting results name 'level1' on Or expression will return a list of all parsed tokens in an And alternative, in prior versions only the first token was returned; enclose contained argument in Group
  edtfParser = level0Expression("level0") ^ level1Expression("level1") ^ level2Expression("level2")
/Users/johnathanclementi/Documents/Arches/Projects/rdm/ENV76/lib/python3.12/site-packages/edtf/parser/grammar.py:278: UserWarning: warn_multiple_tokens_in_named_alternation: setting results name 'level2' on Or expression will return a list of all parsed tokens in an And alternative, in prior versions only the first token was returned; enclose contained argument in Group
  edtfParser = level0Expression("level0") ^ level1Expression("level1") ^ level2Expression("level2")
/Users/johnathanclementi/Documents/Arches/Projects/rdm/ENV76/lib/python3.12/site-packages/pytz/tzinfo.py:27: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
  _epoch = datetime.utcfromtimestamp(0)
System check identified some issues:

WARNINGS:
?: (arches.W001) Cache backend does not support rate-limiting
        HINT: Your cache: django.core.cache.backends.dummy.DummyCache
        Supported caches: ('django.core.cache.backends.memcached.PyLibMCCache', 'django.core.cache.backends.memcached.PyMemcacheCache', 'django.core.cache.backends.redis.RedisCache')
?: (guardian.W001) Guardian authentication backend is not hooked. You can add this in settings as eg: `AUTHENTICATION_BACKENDS = ('django.contrib.auth.backends.ModelBackend', 'guardian.backends.ObjectPermissionBackend')`.
?: (staticfiles.W004) The directory '/Users/johnathanclementi/Documents/Arches/Projects/rdm/arches-lingo/arches_lingo/media/build' in the STATICFILES_DIRS setting does not exist.
?: (staticfiles.W004) The directory '/Users/johnathanclementi/Documents/Arches/Projects/rdm/arches-lingo/node_modules' in the STATICFILES_DIRS setting does not exist.
?: (urls.W005) URL namespace 'admin' isn't unique. You may not be able to reverse all URLs in this namespace
?: (urls.W005) URL namespace 'oauth2' isn't unique. You may not be able to reverse all URLs in this namespace
operation: install
2024-08-01 18:55:52,632 arches.app.utils.task_management ERROR    A celery broker is running, but a celery worker is not available
/Users/johnathanclementi/Documents/Arches/Projects/rdm/arches/arches/management/commands/packages.py:339: DeprecationWarning: The install operation does nothing since Arches 7.6. In Arches 8.0, calling this operation will raise an exception.
  warnings.warn(
sys:1: ResourceWarning: unclosed <socket.socket fd=4, family=2, type=1, proto=6, laddr=('127.0.0.1', 52022), raddr=('127.0.0.1', 5674)>
ResourceWarning: Enable tracemalloc to get the object allocation traceback
(ENV76) johnathanclementi@Johnathans-MacBook-Pro arches-lingo %
```

Reply:
Deprecation warnings are hidden by default to hide them from end users, but this is always a hard judgement call when you're developing a library and your end users are mostly other developers. Here I think someone running `manage.py packages -o install` is an end user, so I'll just elevate it to UserWarning to make sure they see it.

Going forward, other deprecations are likely to surface in project testing / github actions, given that we've turned on deprecation warnings in the github action template.